### PR TITLE
Add the usage and relevant info. of the differentiable head to README

### PR DESCRIPTION
**Pull request by blakechi**
This PR is opened for resolving the issue #143.

- Added a brief introduction of the differentiable head
- Added the usage of the differentiable head
- Added some empirical advises for choosing the hyper parameters
- Added a note for the differentiable head when using multilabel datasets

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,10 +32,10 @@ The examples below provide a quick overview on the various features supported in
 
 `setfit` is integrated with the [Hugging Face Hub](https://huggingface.co/) and provides two main classes:
 
-* `SetFitModel`: a wrapper that combines a pretrained body from `sentence_transformers` and a classification head from `scikit-learn`
+* `SetFitModel`: a wrapper that combines a pretrained body from `sentence_transformers` and a classification head from [`scikit-learn`](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) or [`SetFitHead`](https://github.com/huggingface/setfit/blob/main/src/setfit/modeling.py) (a differentiable head built upon `PyTorch` with similar APIs to `sentence_transformers`).
 * `SetFitTrainer`: a helper class that wraps the fine-tuning process of SetFit.
 
-Here is an end-to-end example:
+Here is an end-to-end example using `scikit-learn`:
```
Comment (lewtun) [inline README.md:38]: nit:
```suggestion
Here is an end-to-end example using a classification head from `scikit-learn`:
```

File: README.md [hunk 2]
```diff
@@ -82,6 +82,74 @@ model = SetFitModel.from_pretrained("lewtun/my-awesome-setfit-model")
...
+# Push model to the Hub
+trainer.push_to_hub("my-awesome-setfit-model")
+
+# Download from Hub and run inference
+model = SetFitModel.from_pretrained("lewtun/my-awesome-setfit-model")
+# Run inference
+preds = model(["i loved the spiderman movie!", "pineapple on pizza is the worst 🤮"]) 
+```
+
+Based on the experiments, `SetFitHead` can achieve similar performance as using `sklearn`'s head. We use `AdamW` as the optimizer and scale down learning rates by 0.5 every 5 epochs. For more details about the experiments, please check out [here](https://github.com/huggingface/setfit/pull/112#issuecomment-1295773537). We recommend using a large learning rate (e.g. `1e-2`) for `SetFitHead` and a small learning rate (e.g. `1e-5`) for the body in your first attempt.
```
Comment (lewtun) [inline README.md:151]:
```suggestion
Based on our experiments, `SetFitHead` can achieve similar performance as using a `scikit-learn` head. We use `AdamW` as the optimizer and scale down learning rates by 0.5 every 5 epochs. For more details about the experiments, please check out [here](https://github.com/huggingface/setfit/pull/112#issuecomment-1295773537). We recommend using a large learning rate (e.g. `1e-2`) for `SetFitHead` and a small learning rate (e.g. `1e-5`) for the body in your first attempt.
```

File: README.md [hunk 3]
```diff
@@ -82,6 +82,74 @@ model = SetFitModel.from_pretrained("lewtun/my-awesome-setfit-model")
...
+# Simulate the few-shot regime by sampling 8 examples per class
+num_classes = 2
+train_dataset = dataset["train"].shuffle(seed=42).select(range(8 * num_classes))
+eval_dataset = dataset["validation"]
+
+# Load a SetFit model from Hub
+model = SetFitModel.from_pretrained(
+    "sentence-transformers/paraphrase-mpnet-base-v2",
+    use_differentiable_head=True,
+    head_params={"out_features": num_classes},
```
Comment (lewtun) [inline README.md:107]: Unrelated to this PR, but I've been wondering if it would be clearer to just have `num_labels` as an arg instead of `head_params`. 

Are there some use cases where it makes sense to expose the full set of `head_params`?

File: README.md [hunk 4]
```diff
@@ -32,10 +32,10 @@ The examples below provide a quick overview on the various features supported in
 
 `setfit` is integrated with the [Hugging Face Hub](https://huggingface.co/) and provides two main classes:
 
-* `SetFitModel`: a wrapper that combines a pretrained body from `sentence_transformers` and a classification head from `scikit-learn`
+* `SetFitModel`: a wrapper that combines a pretrained body from `sentence_transformers` and a classification head from [`scikit-learn`](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) or [`SetFitHead`](https://github.com/huggingface/setfit/blob/main/src/setfit/modeling.py) (a differentiable head built upon `PyTorch` with similar APIs to `sentence_transformers`).
```
Comment (lewtun) [inline README.md:35]: nit:
```suggestion
* `SetFitModel`: a wrapper that combines a pretrained body from `sentence_transformers` and a classification head from either [`scikit-learn`](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) or [`SetFitHead`](https://github.com/huggingface/setfit/blob/main/src/setfit/modeling.py) (a differentiable head built upon `PyTorch` with similar APIs to `sentence_transformers`).
```

**Review by lewtun — APPROVED** [review]
Thanks a lot for adding these clarifications @blakechi 🚀 !

I've left some nits, but otherwise this LGTM :)

File: README.md (hunk 3 above)
Comment (blakechi) [reply README.md:107]: I remember I wanted to align the same APIs as the [sklearn's head](https://github.com/huggingface/setfit/blob/main/tests/test_modeling.py#L56-L71).

I guess the idea was to separate the head's parameters from others? But ya, it's not pretty intuitive.

**Comment by blakechi** [comment]
Just fixed it😃

File: README.md (hunk 3 above)
Comment (lewtun) [reply README.md:107]: Thanks, that makes sense. I'll play around and might consider a refactor in future :)

**Review by lewtun — APPROVED** [review]
Thanks for the fixes @blakechi - this looks great 🚀 !
